src/reducers-actions/orders.ts

```diff
@@ -22,13 +23,53 @@ export const appendOrders = (orders: AuctionElement[]): UpdateOrdersActionType =
   payload: { orders },
 })
 
+export const updateOrders = (orders: AuctionElement[]): UpdateOrdersActionType => ({
+  type: 'UPDATE_ORDERS',
+  payload: { orders },
+})
+
 export const updateOffset = (offset: number): UpdateOffsetActionType => ({
   type: 'UPDATE_OFFSET',
   payload: { offset },
 })
 
 export const INITIAL_ORDERS_STATE = { orders: [], offset: 0 }
 
+/**
+ * When orders are `deleted` from the contract, they are still returned, but with all fields set to zero.
+ * We will not display such orders.
+ *
+ * This function checks whether the order has been zeroed out.
+ * @param order The order object to check
+ */
+function isOrderDeleted(order: AuctionElement): boolean {
+  return (
+    order.buyTokenId === 0 &&
+    order.sellTokenId === 0 &&
+    order.priceDenominator.eq(ZERO) &&
+    order.priceNumerator.eq(ZERO) &&
+    order.validFrom === 0 &&
+    order.validUntil === 0
+  )
+}
+
+function classifyOrders(
+  allOrders: AuctionElement[],
+): { activeOrders: { [id: string]: number }; deletedOrdersIds: Set<string> } {
+  const activeOrders: { [id: string]: number } = {}
+  const deletedOrdersIds = new Set<string>()
+
+  allOrders.forEach((order, index) => {
+    if (isOrderDeleted(order)) {
+      deletedOrdersIds.add(order.id)
+    } else {
+      activeOrders[order.id] = index
```

Comment:
are indices in `Set` guaranteed to be in order?

Reply:
Good point David, although I think it does. 
And that's another weird thing about JS, sets are unordered in any other language, list have order 🤯

![image](https://user-images.githubusercontent.com/2352112/79977277-4a08a080-849e-11ea-8b80-fb0e10b742c3.png)

Reply:
Does not matter in this case. I just want a quick access to identify the ones that should be removed. I'm relying on the existing orders list instead.